### PR TITLE
React to hosting

### PR DESCRIPTION
**Pull request by HaoK**
@davidfowl this looks a bit weird, but I think it works...

cc @divega

### CODE REVIEW

File: src/EntityFramework.Commands/ContextTool.cs [hunk 1]
```diff
@@ -52,6 +52,35 @@ private static DbContext TryCreateContextFromStartup(Type type)
...
+            {
+                private readonly Func<IFeatureCollection, Task> _application;
+
+                public StartInstance(Func<IFeatureCollection, Task> application)
+                {
+                    _application = application;
+                }
+
+                public void Dispose()
+                {
```
Comment (divega) [inline src/EntityFramework.Commands/ContextTool.cs]: Nothing to dispose I guess. Is this what looks weird? :smile:

Comment (HaoK) [inline src/EntityFramework.Commands/ContextTool.cs]: Also needing to write all this infrastructure just to get a Hosting built ApplicationServices out.  Lots of repos are needing this exact usage (creating a hosting built ISP)

**Comment by HaoK** [comment]
Merged 167e85a65d96cd72715e8491c6e45309e9a46b97

File: src/EntityFramework.Commands/ContextTool.cs (hunk 1 above)
Comment (davidfowl) [inline src/EntityFramework.Commands/ContextTool.cs]: We might need to look at this a bit. Seems like we might need to be able to boot the hostingEngine without a server (that's what this is). If you replace the IServerFactory you get the same effect.

Comment (divega) [inline src/EntityFramework.Commands/ContextTool.cs]: Good to hear this is a common need. Is it always about desig-time?

Comment (divega) [inline src/EntityFramework.Commands/ContextTool.cs]: I am going to guess testing is another reason.

Comment (HaoK) [inline src/EntityFramework.Commands/ContextTool.cs]: Yeah mostly its testing I think

**Pull request closed without merging**